mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,427 @@
+                    raw_model, maskers
+            ):
+                explainer = shap.explainers.Additive(
+                    raw_model, maskers, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(sampled_X)
+            else:
+                # fallback to default explainer
+                explainer = shap.Explainer(
+                    predict_fn, sampled_X, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(sampled_X)
+
+        _logger.info(f'Shap explainer {explainer.__class__.__name__} is used.')
+
+        # TODO: seems infer pip req fail when log_explainer.
+        # TODO: The explainer saver is buggy, if `get_underlying_model_flavor` return "unknown",
+        #   then fallback to shap explainer saver, and shap explainer will call `model.save`
+        #   for sklearn model, there is no `.save` method, so error will happen.
+
+        mlflow.shap.log_explainer(
+            explainer,
+            artifact_path=DefaultEvaluator._gen_log_key('explainer', dataset_name)
+        )
+
+
+        def plot_beeswarm():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_beeswarm, run_id, "shap_beeswarm", dataset_name,
+        )
+
+        def plot_summary():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.summary_plot(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_summary, run_id, "shap_summary", dataset_name,
+        )
+
+        def plot_feature_importance():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts,
+            temp_dir,
+            plot_feature_importance,
+            run_id,
+            "shap_feature_importance",
+            dataset_name,
+        )
+
+    def _evaluate_classifier(self, temp_dir, model, X, y, dataset_name, feature_names, run_id, evaluator_config):
+        from mlflow.models.evaluation.lift_curve import plot_lift_curve
+
+        # Note: require labels to be number of 0, 1, 2, .. num_classes - 1
+        label_list = sorted(list(set(y)))
+        assert label_list[0] == 0, "Label values must being at '0'."
+        num_classes = len(label_list)
+
+        y_pred = model.predict(X)
+
+        is_binomial = num_classes <= 2
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+        metrics["example_count"] = len(X)
+
+        # TODO: sum/mean on what data ?
+        #  [P0] Sum: Computes the (weighted) sum of the given values.
+        #  [P0] Mean: Computes the (weighted) mean of the given values.
+
+        if is_binomial:
+            if model.support_predict_proba():
```

Comment:
Adjust the predict_proba interface.
I create another PR for adding it:
https://github.com/mlflow/mlflow/pull/5175

Reply:
Left a comment on #5175. I don't think we can expand the pyfunc prediction interface without going through a design process with the Inference team and discussing how it impacts model serving (MLflow Scoring). Instead, can we create special cases for sklearn and xgboost for now?

Also, what happens if the `model.predict()` method produces probabilities directly? Can we try to detect that?